crates/holochain_persistence_lmdb/src/cas/lmdb.rs

```diff
@@ -0,0 +1,196 @@
+use holochain_json_api::json::JsonString;
+use holochain_persistence_api::{
+    cas::{
+        content::{Address, AddressableContent, Content},
+        storage::ContentAddressableStorage,
+    },
+    error::{PersistenceError, PersistenceResult},
+    reporting::{ReportStorage, StorageReport},
+};
+use rkv::{
+    error::{DataError, StoreError},
+    DatabaseFlags, EnvironmentFlags, Manager, Rkv, SingleStore, StoreOptions, Value,
+};
+use std::{
+    fmt::{Debug, Error, Formatter},
+    path::Path,
+    sync::{Arc, RwLock},
+};
+use uuid::Uuid;
+
+const CAS_BUCKET: &str = "cas";
+const MAX_SIZE_BYTES: usize = 104857600; // TODO: Discuss what this should be, currently 100MB
+
+#[derive(Clone)]
+pub struct LmdbStorage {
+    id: Uuid,
+    store: SingleStore,
+    manager: Arc<RwLock<Rkv>>,
+}
+
+impl Debug for LmdbStorage {
+    fn fmt(&self, f: &mut Formatter) -> Result<(), Error> {
+        f.debug_struct("LmdbStorage").field("id", &self.id).finish()
+    }
+}
+
+impl LmdbStorage {
+    pub fn new<P: AsRef<Path> + Clone>(db_path: P) -> LmdbStorage {
+        let cas_db_path = db_path.as_ref().join("cas").with_extension("db");
+        std::fs::create_dir_all(cas_db_path.clone())
+            .expect("Could not create file path for CAS store");
+
+        let manager = Manager::singleton()
+            .write()
+            .unwrap()
+            .get_or_create(cas_db_path.as_path(), |path: &Path| {
+                let mut env_builder = Rkv::environment_builder();
+                env_builder
+                    // max size of memory map, can be changed later
+                    .set_map_size(MAX_SIZE_BYTES)
+                    // max number of DBs in this environment
+                    .set_max_dbs(1)
+                    // Thes flags make writes waaaaay faster by async writing to disk rather than blocking
+                    // There is some loss of data integrity guarantees that comes with this
+                    .set_flags(EnvironmentFlags::WRITE_MAP | EnvironmentFlags::MAP_ASYNC);
```

Comment:
I believe it opens up the possibility of corrupting the whole database by writing to the wrong places in memory since it doesn't verify before syncing to disk. But Rust should stop us from doing that right?